app/routes.py

```diff
@@ -1,2 +1,328 @@
-from flask import Blueprint
+from flask import Blueprint, request, jsonify
+from werkzeug.wrappers import PlainRequest
+from app import db
+from flask.helpers import make_response
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import date
+import os
+import requests
 
+# sets up blueprints - with details
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# 
+# Wave 1: GET & POST functions - gets data for all tasks in table, and creates new tasks
+# come back and refactor into task_index & independent POST function
+@tasks_bp.route("", methods=["GET","POST"], strict_slashes=False)
+def handle_tasks():
+    if request.method == "GET":
+        task_title_from_url = request.args.get("title")
+        # search for task by title
+        if task_title_from_url:
+            tasks = Task.query.filter_by(title=task_title_from_url)
+        # all tasks
+        else:
+            tasks = Task.query.all()
+    
+        tasks_response = []
+
+        for task in tasks:
+            tasks_response.append(task.to_json())
+
+        # Wave 2 ascending/descending logic for GET method
+        if "asc" in request.full_path:
+            sorted_ascending = sorted(tasks_response, key=lambda x: x.get("title"))
+            return jsonify(sorted_ascending)
+        
+        elif "desc" in request.full_path:
+            sorted_descending = sorted(tasks_response, key=lambda x: x.get("title"), reverse=True)
+            return jsonify(sorted_descending)
+
+        return jsonify(tasks_response), 200
```

Comment:
👍 this works great! We could shorten this and return only one response:

```suggestion
        if "asc" in request.full_path:
            tasks_response = sorted(tasks_response, key=lambda x: x.get("title"))
        
        elif "desc" in request.full_path:
            tasks_response = sorted(tasks_response, key=lambda x: x.get("title"), reverse=True)

        return jsonify(tasks_response), 200
```